core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -220,7 +221,17 @@ public void releaseManagedChannel(ChannelKey channelKey, long shutdownTimeoutMs)
    * @return the created channel
    */
   private ManagedChannel createManagedChannel(ChannelKey channelKey) {
-    NettyChannelBuilder channelBuilder = NettyChannelBuilder.forAddress(channelKey.mAddress);
+    NettyChannelBuilder channelBuilder;
+    // Use constructor that does auto resolving if given an unresolved inet address.
```

Comment:
nit: update comment to mentioned delayed dns lookup to handle changes

Reply:
Updated the doc but there is nothing delayed here. From gRPC's perspective it either resolves or not. Delay is caused by master inquire client constnatly handing over unresolved address.